app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php

```diff
@@ -127,10 +127,25 @@ public function initForm()
             Mage::helper('adminhtml/addresses')
                 ->processStreetAttribute($attributes['street']);
         }
+
+        $customerStoreId = null;
+        if ($customer->getId()) {
+            $customerStoreId = Mage::app()->getWebsite($customer->getWebsiteId())->getDefaultStore()->getId();
+        }
+
         foreach ($attributes as $attribute) {
             /* @var $attribute Mage_Eav_Model_Entity_Attribute */
             $attribute->setFrontendLabel(Mage::helper('customer')->__($attribute->getFrontend()->getLabel()));
             $attribute->unsIsVisible();
+            
+            /* Sets store id to attribute if select or multiselect because values are loaded store specific.
+               If not set values of default store are loaded which can be undesirable (e.g. country list in customer
+               addresses). */
+            /** @see Mage_Adminhtml_Block_Widget_Form::_setFieldset() lines 208 & 210 */
+            $frontendInputType = $attribute->getFrontend()->getInputType();
+            if ($customerStoreId !== null && ($frontendInputType === 'select' || $frontendInputType === 'multiselect')) {
+                $attribute->setStoreId($customerStoreId);
+            }
```

Comment:
```suggestion
            $attribute->setStoreId($customerStoreId); // For country list
```
I think it's unnecessary to check for `null` and input type.